include/lbann/callbacks/check_gradients.hpp

```diff
@@ -56,24 +57,32 @@ class check_gradients : public callback_base {
    *  @param error_on_failure   Whether to throw an exception for
    *                            large gradient errors.
    */
-  check_gradients(DataType step_size = DataType(0),
-                                 bool verbose = false,
-                                 bool error_on_failure = false);
+  check_gradients(std::set<execution_mode> modes = {},
```

Comment:
The documentation should say what the default (empty set) does.

Reply:
Done.